### PR TITLE
4879 r42 phn lookup

### PR DESCRIPTION
**Pull request by weskubo-cgi**
This is the implementation of R42 including integration with the backend service. There's a lot going on here since this is the first time we've done a roundtrip to this service. I'm not huge on having both the RapidDefaults and MSHDefaults classes...I'll probably combine them into a single Properties class that can be used wherever application properties are required but injection isn't used. But it works for now.

Also, please ignore anything related to R41/InquirePHN as it's still in progress. I branched from there to start R42 since I was blocked on the R41 format.

### CODE REVIEW

File: backend/src/main/resources/application.yaml [hunk 1]
```diff
@@ -20,4 +20,9 @@ v2:
...
     sendingFacility: HN-WEB
     receivingFacility: BC00001013
-    processingID: D
\ No newline at end of file
+    processingID: D
+
+# RAPID fixed width message construction
+rapid:
+  header: 
+    organization: "00000010"
```
Comment (tschia) [inline backend/src/main/resources/application.yaml:28]: I think this might actually need to come from the token, the same way as we've done for V2

Comment (weskubo-cgi) [reply backend/src/main/resources/application.yaml:28]: I updated the converter to use the existing code to grab it from the token. There will be some further changes in the R41 branch to remove the RapidDefaults code entirely.

File: frontend/src/services/BaseService.js [hunk 2]
```diff
@@ -3,28 +3,23 @@ import axios from 'axios';
...
-      eligibility: '/eligibility',
-      checkEligibility: '/eligibility/check-eligibility',
-      checkMspCoverageStatus: '/eligibility/check-msp-coverage-status',
-      phnEnquiry: '/eligibility/phnEnquiry'
-    }
+  eligibility: {
+    checkEligibility: '/eligibility/check-eligibility',
+    checkMspCoverageStatus: '/eligibility/check-msp-coverage-status',
+    inquirePhn: '/eligibility/inquire-phn',
+    lookupPhn: '/eligibility/lookup-phn'
```
Comment (daveb-hni) [inline frontend/src/services/BaseService.js:10]: missing trailing comma

File: frontend/src/util/validators.js [hunk 3]
```diff
@@ -4,12 +4,28 @@ import { helpers } from '@vuelidate/validators'
 
 /**
  * Validates that the PHN matches the accepted format.
+ * This assumes the PHN is optional and an empty value won't
+ * case validation failure.
```
Comment (daveb-hni) [inline frontend/src/util/validators.js:8]: typo case => cause

**Review by daveb-hni — COMMENTED** [review]
Looks good, few very minor comments.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/exception/ExceptionType.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package ca.bc.gov.hlth.hnweb.exception;
+
+public enum ExceptionType {
+	DOWNSTREAM_FAILURE("Could not connect to downstream service"),
+	GENERAL("An error has occured"),
+	SSL_FAILURE("SSL Context for could not be built, application will not start");
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/exception/ExceptionType.java:6]: Yeah, I added this for that purpose but forgot to use it.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/config/WebClientConfig.java [hunk 5]
```diff
@@ -84,8 +117,8 @@ private SslContext getSSLContext() throws HNWebException {
 			        .keyManager(keyManagerFactory)
 			        .build();
 		} catch (IOException | CertificateException | UnrecoverableKeyException | KeyStoreException | NoSuchAlgorithmException e) {
-	        logger.error("Error creating Enroll Subscriber SSL Context due to {}.", e.toString());
-	        throw new HNWebException("SSL Context for Enroll Subscriber could not be built, application will not start", e.getCause());
+	        logger.error("Error creating SSL Context for {} due to {}.", url, e.toString());
+	        throw new HNWebException("SSL Context for could not be built, application will not start", e.getCause());
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/config/WebClientConfig.java:121]: Updated logging and fixed to use an ExceptionType instead.

**Review by daveb-hni — APPROVED** [review]
All good

**Pull request merged**